Add review snippet to structured data

robinandeer opened the pull request:
## What?

Add review snippet to structured data for our Trustpilot score:
https://developers.google.com/search/docs/advanced/structured-data/review-snippet

## Why?

Display in Google search results.

## Next step

We can fetch the information from this endpoint:

https://api.trustpilot.com/v1/business-units/5b62ebf41788620001d3c4ae?apikey=XXXXX

However, it requires an API key so we should perform that request from the backend.

Commit:
Add review snippet to structured data

styrelius reviewed (commented):
If we're not going to use the TrustPilot API to get the current score since that's somewhat a bigger project that involves back-end and so on, I would suggest that the `ratingValue`, the `bestRating` and the `reviewCount` should be added/modified from Storyblok, like we do with the organization description. 
If we have to change it in the actual codebase we most certainly will forget to and the numbers will very soon be very wrong 😬

gustaveen commented on src/utils/structuredData.ts:
```diff
@@ -89,3 +88,21 @@ export const structuredOrganization = ({
+  '@context': 'http://schema.org',
+  '@type': 'AggregateRating',
+  itemReviewed: structuredDataOrganization({ description }),
+  ratingValue: '4.8',
```
Since we have a hardcoded value of our Trustpilot score on the site today I think it's fine to have it here as well for the time being :)

gustaveen reviewed (commented):
Yes, since we have a hardcoded value of our Trustpilot score on the site today I think it's fine to have it here as well for the time being or as you suggested added to Storyblok 👍 Best rating will always be 5 I assume so that can be omited probably

robinandeer commented:
Right, it's probably better to store this information in Storyblok. I can look into that.

Commit:
Store Trustpilot review score in Storyblok

Add script to update Trustpilot score.
Update README with information about new script.
Use Trustpilot score from Storyblok in meta data.

Commit:
Update component definitions

robinandeer commented:
@gustaveen @styrelius I've updated the implementation now.

I investigated directly calling the Trustpilot API to get the values initially. However, the values don't need to be updated very often so I believe it's necessary.

My solution caches the values in Storyblok like we talked about ⬆️ 

I've also added a script that calls the Trustpilot API and updates Storyblok automatically.

Next step could potentially be to run that script on a schedule or e.g. when we deploy the site.

styrelius reviewed (commented):
Very cool! 🙌 🙌  I think we need some more documentation on how to run this though. I've been struggling for a good amount of time now with different ID's from Storyblok and TrustPilot and have still not been able to run the script successfully. 😔

gustaveen reviewed (commented):
Great update! 🎉 Works great! Just a question really about the business unit ID, is it unique for all markets or is it the same? 
Minor suggesion, maybe group the Trustpilot settings under a Trustpilot folder? 
<img width="1014" alt="Screenshot 2021-08-24 at 14 19 20" src="https://user-images.githubusercontent.com/6661511/130615122-23dd0249-3b92-4835-ad88-9e1ae2339050.png">

Commit:
Accept business unit on command line

Update docs in README.
Remove trustpilot business unit from Storyblok.

robinandeer commented:
@gustaveen 
> Could there be a way to get the Global stories dynamically instead of specifying page ids? :) But that might be overkill.

Yeah there probably is but I think we need to integrate with the delivery API then as well. I think if we integrate with CI and run it automatically, this solution might be easier to maintain. Does that make sense?

robinandeer commented:
@gustaveen 
> Great update! 🎉 Works great! Just a question really about the business unit ID, is it unique for all markets or is it the same?
> Minor suggesion, maybe group the Trustpilot settings under a Trustpilot folder?

I followed @styrelius suggestion and moved the business unit param to the command line. It's not something editors should need to worry about so I don't think it belongs in Storyblok on second thought.

Perhaps we don't have different business units for each market. However, do we have any place to store such cross-market global values?

styrelius commented on README.md:
```diff
@@ -165,3 +165,22 @@ a) Running 2 commands isn't too bad right? b) 3 real reasons:
+**Flags**
+
+- `--space <id>` Storyblok Space ID to fetch and update.
+- `--page <id>` Storyblok Page ID for the locale-specific "global" story.
```
😍

styrelius commented:
I agree with @gustaveen that it would be neat to group the Trustpilot settings under a Trustpilot folder in Storyblok - to make it clearer what it is for the editors of Storyblok. Also maybe add a comment that they shouldn't touch these fields since they are updated from the TrustPilot API?

gustaveen commented on README.md:
````diff
@@ -168,11 +168,10 @@ a) Running 2 commands isn't too bad right? b) 3 real reasons:
 
 ```sh
-yarn update-trustpilot-score --space 123714 --page 67792582
+yarn update-trustpilot-score --space 123714 --page 67792582 --businessunit 5b62ebf41788620001d3c4ae
````
Great! Makes perfect sense 🙌

gustaveen reviewed (commented):
> However, do we have any place to store such cross-market global values?

No we don't, so that's what I realised might be nice to have longterm, if there will be a bunch of repeated settings on all global setting pages going forward :)

gustaveen reviewed (dismissed):
🙌

Commit:
Remove deleted storyblok field

styrelius reviewed (commented):
Nice! Still think that it's a good idea like Gustav wrote to group these review things under a **Trustpilot** folder, like we do with e.g. **Structured Data** (i.e. add a `section` to `storyblok/components.json`), and also have a little description for each of these input fields so that the editors of Storyblok know that they shouldn't manually update these fields. By the way could they be "locked"/disabled in some way so can't edit them manually?

Also it seems like these fields are lacking the `pos` property? 

![image](https://user-images.githubusercontent.com/42962286/130943384-a11ca910-de48-4d23-816f-0096c040b9ea.png)

Commit:
Add description to new storyblok fields

Group fields under folder in editor.

gustaveen reviewed (approved):
⭐️
